src/db/mod.rs

```diff
@@ -295,4 +296,19 @@ impl Database {
             )
             .await
     }
+
+    pub(crate) fn deserialize_str_from_map<'de, D>(
```

Comment:
This doesn't have anything to do with the database type, so I don't think it should be inside the `Database` impl block. Please move it to the `bson_util` module as a top-level function.

Reply:
fixed